### PR TITLE
JS-OOP

### PR DESCRIPTION
**Pull request by Zakkarat**
### Task Frogger

[Demo](https://zakkarat.github.io/frontend-nanodegree-arcade-game/) |
[Code base](https://github.com/Zakkarat/frontend-nanodegree-arcade-game)

The code is submitted in its own sub-directory and in a dedicated feature branch.
Please, review.

### CODE REVIEW

**Comment by Oodmincheg** [comment]
Hello! Very good job! I see you imlemented addional functionality and it's great! Maybe it'd be nice if you add come message about end of the game? Maybe little delay and message, cause now it's very confusing situation: player just get his new high score(it it occur) and game start again immidiatly. But  don't insist :)

**Comment by Oodmincheg** [comment]
And can you left in PR only file that was changed by you?(i think it's one ore maybe two js files). Thank you :) Ask me any questions here or in the Telegram.

File: submissions/Zakkarat/JS-OOP/js/app.js [hunk 1]
```diff
@@ -0,0 +1,142 @@
+const startX = 200;
+const startY = 380;
+const stepX = 100;
+const stepY = 80;
```
Comment (Oodmincheg) [inline submissions/Zakkarat/JS-OOP/js/app.js]: * Requirements re **Constants**:
   * [ ] all numbers like block dimensions, initial locations are defined as constants
   * [ ] there are core constants and derived constants
         (e.g. `const FIELD_WIDTH = BLOCK_WIDTH * BLOCKS_NUMBER;`)
   * [ ] arrays of constants are also constants
         (e.g. `const INITIAL_POSITIONS = [1,2,3,4].map(rowNumber => rowNumber * BLOCK_HEIGHT);`)
   * [ ] const objects help organizing and structure const data even better
         (e.g. `const PLAYER_CONF = { initialPosition: {x: 1, y: 5}, sprite: '...', ...etc... };`

File: submissions/Zakkarat/JS-OOP/js/app.js [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    this.y = y;
+    this.sprite = 'images/enemy-bug.png';
+};
+
+function speedRand() {
+  let speed = Math.floor(Math.random() * (375 - 200 ) + 200);
+  return speed;
+}
+function positionRand() {
+  let position = Math.floor(Math.random() * (-225))
```
Comment (Oodmincheg) [inline submissions/Zakkarat/JS-OOP/js/app.js]: Move 'magic numbers' to constants with meaningfull names, please.

File: submissions/Zakkarat/JS-OOP/js/app.js [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+  return position;
+}
+
+Enemy.prototype.colision = function() {
+  return (Math.abs(this.x - player.x) < 60 && Math.abs(this.y - player.y) < 70)
+}
+
+Enemy.prototype.update = function(dt) {
+    this.x += this.speed * dt;
+    if (this.x > 500) {
```
Comment (Oodmincheg) [inline submissions/Zakkarat/JS-OOP/js/app.js]: 'Magiiiiic' :) Fix them in all your code, please.

File: submissions/Zakkarat/JS-OOP/js/app.js [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+          }
+            break;
+    }
+};
+// Now instantiate your objects.
+// Place all enemy objects in an array called allEnemies
+// Place the player object in a variable called player
+let player = new Player(startX, startY);
+
+let allEnemies = [new Enemy(60), new Enemy(140), new Enemy(230)];
```
Comment (Oodmincheg) [inline submissions/Zakkarat/JS-OOP/js/app.js]: * [ ] arrays of constants are also constants
         (e.g. `const INITIAL_POSITIONS = [1,2,3,4].map(rowNumber => rowNumber * BLOCK_HEIGHT);`)

File: submissions/Zakkarat/JS-OOP/js/app.js [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+  if (lives == 0){
+    lives = 3;
+    highScr();
+  }
+  scr[0].textContent = lives;
+}
+function updScore() {
+  score++;
+  scr[2].textContent = score;
+}
```
Comment (Oodmincheg) [inline submissions/Zakkarat/JS-OOP/js/app.js]: And there is a very good opportunity level up your OOP skill :) Please think of the `score`, `lives`, `updScore`, `updLives` how about properties and methods of `Player` class. Try to implement them in this way. If you get stuck ask me any questions here on in the Telegram.

**Comment by Oodmincheg** [comment]
Very nice, man. Now look at the comments :)

File: submissions/Zakkarat/JS-OOP/js/app.js (hunk 1 above)
Comment (Zakkarat) [reply submissions/Zakkarat/JS-OOP/js/app.js]: Done!

File: submissions/Zakkarat/JS-OOP/js/app.js (hunk 5 above)
Comment (Zakkarat) [reply submissions/Zakkarat/JS-OOP/js/app.js]: Done!

File: submissions/Zakkarat/JS-OOP/js/app.js [hunk 6]
```diff
@@ -49,49 +67,78 @@ const Player = function(x, y) {
...
+  allowMove = true;
+});
 // Now instantiate your objects.
 // Place all enemy objects in an array called allEnemies
 // Place the player object in a variable called player
-let player = new Player(startX, startY);
+let player = new Player(PlayerConf.startX, PlayerConf.startY);
 
-let allEnemies = [new Enemy(60), new Enemy(140), new Enemy(230)];
+let allEnemies = [new Enemy(EnemyConf.minStartY[0]), new Enemy(EnemyConf.minStartY[1]), new Enemy(EnemyConf.minStartY[2])];
```
Comment (Oodmincheg) [inline submissions/Zakkarat/JS-OOP/js/app.js]: let allEnemies = minStartY.map(startY => new Enemy(startY)) better I thnik

Comment (Zakkarat) [reply submissions/Zakkarat/JS-OOP/js/app.js]: Thank you. Done!

File: submissions/Zakkarat/JS-OOP/js/app.js [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+        case 'down':
+          if(this.y < Field.minHeight && allowMove) {
+            this.y += PlayerConf.stepY;
+          }
+            break;
+    }
+};
+Player.prototype.updLive = function() {
+    this.lives--;
+    if (this.lives == 0){
```
[the remark this answers is not in the document]
Comment (Zakkarat) [reply submissions/Zakkarat/JS-OOP/js/app.js]: Yep! Thank you. Now its done!

**Comment by Oodmincheg** [comment]
Very nice :) LGTM. @Roka20012  and for you?

**Comment by Roka20012** [comment]
@Oodmincheg LGTM too

**Pull request merged**